packages/react-router/__tests__/path-matching-test.tsx

```diff
@@ -458,31 +458,97 @@ describe("path matching with optional dynamic segments", () => {
   });
 
   test("optional params at the end of the path", () => {
+    let manualRoutes = [
+      {
+        path: "/nested",
+      },
+      {
+        path: "/nested/:one",
+      },
+      {
+        path: "/nested/:one/:two",
+      },
+      {
+        path: "/nested/:one/:two/:three",
+      },
+      {
+        path: "/nested/:one/:two/:three/:four",
+      },
+    ];
     let routes = [
       {
-        path: "/nested/:one?/:two?",
+        path: "/nested/:one?/:two?/:three?/:four?",
       },
     ];
```

Comment:
I updated some of these complex tests to include the equivalent manual route definitions and am asserting against both set of routes to ensure we're consistent with what the user would otherwise do manually.